### PR TITLE
github: Add PR Template

### PR DESCRIPTION
**Pull request by davids5**
@xiaoxiang781216 - Sorry this is so long overdue. Please add the links to running checkpatch in the 2 places indicated or just add the the `commands`

### CODE REVIEW

**Comment by davids5** [comment]
Here is a demo view https://github.com/nuttx-to-asf/incubator-nuttx/compare/master-pr-add-PR-template...nuttx-to-asf:master-pr-add-PR-template-test?expand=1

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+## Title guidelines
+
+Following the guidelines for writting good commit messages (https://chris.beams.io/posts/git-commit/) and creating a meaningful title is key in effective team communication.
+
+**do's**
+- stm32h7:Add SDMMC Support
+- nsh:Separate `source` and `sh` for POSIX compliance
+- nxstyle:Fixed Camel case detection
+- drivers/serial:Fixed style violation
+
```
Comment (patacongo) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: The English language requires two spaces following a colon.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# Pull Request Template
+
+## Title guidelines
+
+Following the guidelines for writting good commit messages (https://chris.beams.io/posts/git-commit/) and creating a meaningful title is key in effective team communication.
+
+**do's**
+- stm32h7:Add SDMMC Support
```
Comment (davids5) [inline .github/PULL_REQUEST_TEMPLATE.md:8]:
```suggestion
- stm32h7:  Add SDMMC Support
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- nsh:Separate `source` and `sh` for POSIX compliance
```
Comment (davids5) [inline .github/PULL_REQUEST_TEMPLATE.md:9]:
```suggestion
- nsh:  Separate `source` and `sh` for POSIX compliance
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+- nxstyle:Fixed Camel case detection
```
Comment (davids5) [inline .github/PULL_REQUEST_TEMPLATE.md:10]:
```suggestion
- nxstyle:  Fixed Camel case detection
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- drivers/serial:Fixed style violation
```
Comment (davids5) [inline .github/PULL_REQUEST_TEMPLATE.md:11]:
```suggestion
- drivers/serial:  Fixed style violation
```

**Comment by davids5** [comment]
@xiaoxiang781216  Please use the suggested changes and add the documentation where indicated for running the spelling check and and NX style then we can squash and merge this

**Comment by davids5** [comment]
> Should we squash the commit into one?

@xiaoxiang781216 Done! Please add the commands to line https://github.com/apache/incubator-nuttx/pull/826/files#diff-195a635ad245ded9076330e31134bd80R68 and 60

**Pull request merged**